qiskit/quantum_info/operators/symplectic/sparse_pauli_op.py

```diff
@@ -0,0 +1,564 @@
+            incompatible dimensions.
+        """
+        # pylint: disable=invalid-name
+        if qargs is None:
+            qargs = getattr(other, 'qargs', None)
+
+        if not isinstance(other, SparsePauliOp):
+            other = SparsePauliOp(other)
+
+        # Validate composition dimensions and qargs match
+        self._get_compose_dims(other, qargs, front)
+
+        # Implement composition of the Pauli table
+        x1, x2 = PauliTable._block_stack(self.pauli.X, other.pauli.X)
+        z1, z2 = PauliTable._block_stack(self.pauli.Z, other.pauli.Z)
+        c1, c2 = PauliTable._block_stack(self.coeffs, other.coeffs)
+
+        if qargs is not None:
+            ret_x, ret_z = x1.copy(), z1.copy()
+            x1 = x1[:, qargs]
+            z1 = z1[:, qargs]
+            ret_x[:, qargs] = x1 ^ x2
+            ret_z[:, qargs] = z1 ^ z2
+            pauli = np.hstack([ret_x, ret_z])
+        else:
+            pauli = np.hstack((x1 ^ x2, z1 ^ z2))
+
+        # Take product of coefficients and add phase correction
+        coeffs = c1 * c2
+        # We pick additional phase terms for the products
+        # X.Y = i * Z, Y.Z = i * X, Z.X = i * Y
+        # Y.X = -i * Z, Z.Y = -i * X, X.Z = -i * Y
+        if front:
+            plus_i = (x1 & ~z1 & x2 & z2) | (x1 & z1 & ~x2 & z2) | (~x1 & z1 & x2 & ~z2)
+            minus_i = (x2 & ~z2 & x1 & z1) | (x2 & z2 & ~x1 & z1) | (~x2 & z2 & x1 & ~z1)
+        else:
+            minus_i = (x1 & ~z1 & x2 & z2) | (x1 & z1 & ~x2 & z2) | (~x1 & z1 & x2 & ~z2)
+            plus_i = (x2 & ~z2 & x1 & z1) | (x2 & z2 & ~x1 & z1) | (~x2 & z2 & x1 & ~z1)
+        coeffs *= 1j ** np.array(np.sum(plus_i, axis=1), dtype=int)
+        coeffs *= (-1j) ** np.array(np.sum(minus_i, axis=1), dtype=int)
+        return SparsePauliOp(pauli, coeffs)
+
+    def dot(self, other, qargs=None):
+        """Return the composition channel self∘other.
+
+        Args:
+            other (SparsePauliOp): an operator object.
+            qargs (list or None): a list of subsystem positions to compose other on.
+
+        Returns:
+            SparsePauliOp: The composed operator.
+
+        Raises:
+            QiskitError: if other cannot be converted to an Operator or has
+            incompatible dimensions.
+        """
+        return self.compose(other, qargs=qargs, front=True)
+
+    def tensor(self, other):
+        """Return the tensor product operator self ⊗ other.
+
+        Args:
+            other (SparsePauliOp): a operator subclass object.
+
+        Returns:
+            SparsePauliOp: the tensor product operator self ⊗ other.
+
+        Raises:
+            QiskitError: if other cannot be converted to a SparsePauliOp
+                         operator.
+        """
+        if not isinstance(other, SparsePauliOp):
+            other = SparsePauliOp(other)
+        pauli = self.pauli.tensor(other.pauli)
+        coeffs = np.concatenate([self.coeffs, other.coeffs])
```

Comment:
Are you sure about this? I thought it was `np.outer(other.coeffs, self.coeffs).reshape(-1)` (or something, not sure if the ordering is right there)

Reply:
Not sure about any of these, still need to write tests

Reply:
Yep, it should have been `np.kron`